src/ophys_etl/modules/roi_cell_classifier/compute_classifier_artifacts.py

```diff
@@ -0,0 +1,275 @@
+                                  avg_img=avg_img,
+                                  corr_img=corr_img,
+                                  exp_id=exp_id,
+                                  out_dir=out_dir)
+
+        self.logger.info(f"Created ROI artifacts in {time.time()-t0:.0f} "
+                         "seconds.")
+
+    def get_max_avg(self,
+                    video_path: pathlib.Path) -> Dict[str, np.ndarray]:
+        """Compute the mean and max projection from a movie on disk.
+
+        Parameters
+        ----------
+        video_path : pathlib.Path
+            Path to HDF5 file containing full movie. Movie is assume to be
+            stored in a dataset named "data".
+
+        Returns
+        -------
+        projections : Dict[str, np.ndarray]
+            Dictionary of image projects of the movie.
+
+            "mean" : Mean projection of the movie.
+            "max" : Max projection of the movie.
+        """
+        with h5py.File(video_path, "r") as in_file:
+            video_data = in_file["data"][()]
+        max_img = np.max(video_data, axis=0)
+        avg_img = np.mean(video_data, axis=0)
+
+        return {"max": max_img, "avg": avg_img}
+
+    def write_thumbnails(self,
+                         extract_roi: ExtractROI,
+                         max_img: np.ndarray,
+                         avg_img: np.ndarray,
+                         corr_img: np.ndarray,
+                         out_dir: pathlib.Path,
+                         exp_id: int):
+        """Compute image cutout artifacts for an ROI.
+
+        Parameters
+        ----------
+        extract_roi : ophys_etl.types.ExtractROI
+            ROI containing bounding box size and location.
+        max_img : np.ndarray
+            Max projection of movie.
+        avg_img : np.ndarray
+            Mean projection of movie.
+        corr_img : np.ndarray
+            Correlation projection of movie.
+        out_dir : pathlib.Path
+            Output path to write png files to.
+        exp_id : int
+            Id of experiment where these ROIs and images come from.
+        """
+        desired_shape = (self.args['cutout_size'], self.args['cutout_size'])
+
+        ophys_roi = extract_roi_to_ophys_roi(extract_roi)
+        pixel_array = ophys_roi.global_pixel_array.transpose()
+
+        # Compute center of cutout from ROI bounding box.
+        if extract_roi['height'] % 2 == 0:
+            center_row = extract_roi['y'] + extract_roi['height'] // 2
+        else:
+            center_row = extract_roi['y'] + extract_roi['height'] // 2 + 1
+        if extract_roi['width'] % 2 == 0:
+            center_col = extract_roi['x'] + extract_roi['width'] // 2
+        else:
+            center_col = extract_roi['x'] + extract_roi['width'] // 2 + 1
+
+        # Get size of cutout.
+        r0 = max(0, center_row - desired_shape[0] // 2)
+        r1 = min(max_img.shape[0], center_row + desired_shape[0] // 2)
```

Comment:
Please add a comment what the `min` and `max` functions are doing here.

Reply:
Check out the new code.